x/dex/keeper/limit_order_tranche.go

```diff
@@ -1,11 +1,88 @@
 package keeper
 
 import (
+	"context"
+
 	"github.com/NicholasDotSol/duality/x/dex/types"
 	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+type LimitOrderTranche struct {
+	Tranche           *types.LimitOrderTranche
+	Tick              *types.Tick
+	PriceTakerToMaker sdk.Dec
+	PriceMakerToTaker sdk.Dec
+}
+
+func NewLimitOrderTranche(
+	tick *types.Tick,
+	tranche *types.LimitOrderTranche,
+	priceMakerToTaker sdk.Dec,
+	priceTakerToMaker sdk.Dec,
+) *LimitOrderTranche {
+	return &LimitOrderTranche{
+		Tranche:           tranche,
+		Tick:              tick,
+		PriceTakerToMaker: priceTakerToMaker,
+		PriceMakerToTaker: priceMakerToTaker,
+	}
+}
+
+func (t *LimitOrderTranche) Swap(maxAmountTaker sdk.Int) (
+	inAmount sdk.Int,
+	outAmount sdk.Int,
+	initedTick *types.Tick,
+	deinitedTick *types.Tick,
+) {
+	tranche := t.Tranche
+	reservesTokenOut := &tranche.ReservesTokenIn
+	fillTokenIn := &tranche.ReservesTokenOut
+	totalTokenIn := &tranche.TotalTokenOut
```

Comment:
Why are we flipping the variable names here (reservesTokenOut, fillTokenIn, totalTokenIn) ?

Reply:
I feel like these should have been renamed, going to double check. Perspective shifts are an unfortunate part of the previous naming scheme, where my: "me/you" is your "you/me". Can extend the analogy to trading "in" for the one person on a trade is "out" for the counterparty. I'd like to move away from subjective names and towards objective names like "maker"/"taker" that don't shift with regard to who you're talking to.